basemap/utils/utils.js

```diff
@@ -216,6 +227,56 @@ function mergeInterpolatedDirective(directives, property, alias, value) {
     }
 }
 
+function mergeInterpolatedDupletColorDirective(directives, property, alias, startZoom, endZoom, fallback) {
+    let cases = []
```

Comment:
What about:

```javascript
let cases = directives.map((rule) => ...)
```

It is better to avoid side effects when doing functional programming.

Reply:
Definitely better.